### PR TITLE
Entity safety API

### PR DESCRIPTION
**Pull request by TheMode**
The goal is to change current methods signature to use the Acquirable API instead of raw entity objects

### CODE REVIEW

File: src/main/java/net/minestom/server/Viewable.java [hunk 1]
```diff
@@ -101,6 +101,6 @@ default void sendPacketToViewersAndSelf(@NotNull ServerPacket packet) {
      * @return the audiences
      */
     default @NotNull Iterable<? extends Audience> getViewersAsAudiences() {
-        return this.getViewers();
+        return this.getViewers().unwrap().collect(Collectors.toList());
```
Comment (kezz) [inline src/main/java/net/minestom/server/Viewable.java:104]: I'm not sure about this, the only thing audience will ever do is directly send packets - is there a way we could access this directly without having to unwrap?

**Review by kezz — COMMENTED** [review]
I'll give it a proper look over some point this week, but it feels like there's a lot of unwrapping happening for simple packet sending, which should be thread safe right?

File: src/main/java/net/minestom/server/Viewable.java (hunk 1 above)
Comment (TheMode) [reply src/main/java/net/minestom/server/Viewable.java:104]: `unwrap` means that there is no acquisition happening

Comment (kezz) [reply src/main/java/net/minestom/server/Viewable.java:104]: Yeah but there must be a better way than going from a collection to a stream and back to a list every time the use just wants to access thread safe methods like this. Could we use the collector view class for this?

Comment (TheMode) [reply src/main/java/net/minestom/server/Viewable.java:104]: We could also use `#iterator` https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/BaseStream.html#iterator()
it does however add an ugly warning. And I do not expect this to have a huge performance impact, could be wrong

**Pull request closed without merging**